contracts/bonding/BondingCheckpoints.sol

```diff
@@ -0,0 +1,393 @@
+     * @dev In the case of delegators it is the amount they are delegating to a transcoder, while for transcoders this
+     * includes all the stake that has been delegated to them (including self-delegated).
+     * @param _account The account whose bonding state we want to get.
+     * @param _round The round for which we want to get the bonding state. Normally a proposal's vote start round.
+     * @return amount The active stake of the account at the given round including any accrued rewards. In case of
+     * transcoders this also includes all the amount delegated towards them by other delegators.
+     * @return delegateAddress The address the account delegated to. Will be equal to _account in case of transcoders.
+     */
+    function getBondingStateAt(address _account, uint256 _round)
+        public
+        view
+        virtual
+        returns (uint256 amount, address delegateAddress)
+    {
+        BondingCheckpoint storage bond = getBondingCheckpointAt(_account, _round);
+
+        delegateAddress = bond.delegateAddress;
+        bool isTranscoder = delegateAddress == _account;
+
+        if (bond.bondedAmount == 0) {
+            amount = 0;
+        } else if (isTranscoder) {
+            // Address is a registered transcoder so we use its delegated amount. This includes self and delegated stake
+            // as well as any accrued rewards, even unclaimed ones
+            amount = bond.delegatedAmount;
+        } else {
+            // Address is NOT a registered transcoder so we calculate its cumulative stake for the voting power
+            amount = delegatorCumulativeStakeAt(bond, _round);
+        }
+    }
+
+    /**
+     * @dev Gets the checkpointed bonding state of an account at a round. This works by looking for the last checkpoint
+     * at or before the given round and using the checkpoint of that round. If there hasn't been checkpoints since then
+     * it means that the state hasn't changed.
+     * @param _account The account whose bonding state we want to get.
+     * @param _round The round for which we want to get the bonding state.
+     * @return The {BondingCheckpoint} pointer to the checkpoints storage.
+     */
+    function getBondingCheckpointAt(address _account, uint256 _round)
+        internal
+        view
+        returns (BondingCheckpoint storage)
+    {
+        if (_round > clock()) {
+            revert FutureLookup(_round, clock());
+        }
+
+        BondingCheckpointsByRound storage checkpoints = bondingCheckpoints[_account];
+
+        // Most of the time we will be calling this for a transcoder which checkpoints on every round through reward().
+        // On those cases we will have a checkpoint for exactly the round we want, so optimize for that.
+        BondingCheckpoint storage bond = checkpoints.data[_round];
+        if (bond.bondedAmount > 0) {
+            return bond;
+        }
+
+        if (checkpoints.startRounds.length == 0) {
```

Comment:
If there are no checkpoints (i.e. startRounds.length == 0) and L255 evaluates to false then it looks like we'll execute L262 which would revert because startRounds.length == 0? This would mean that calling this function for an address without a checkpoint yet (which would be the case immediately post-upgrade for any address that doesn't execute a tx for a bonding action [1]) would cause a revert.

[1] Unless there is a mechanism that guarantees the address has a checkpoint (see my other question about the use of BondingManager.checkpointBondingState()) or that can handle addresses without a checkpoint yet.

Reply:
Fixed on https://github.com/livepeer/protocol/pull/614/commits/d5062fe2db7d5037988c2f1a5989886d7ce087d3
Now we just return 0 balance for uncheckpointed state